src/libYARP_OS/include/yarp/os/Carrier.h

```diff
@@ -520,6 +522,14 @@ class YARP_OS_API yarp::os::Carrier : public Connection
         YARP_UNUSED(params);
     }
 
+    /**
+     * Create new Face object that the carrier needs.
+     *
+     */
+    virtual yarp::os::Face* createFace(void)
```

Comment:
I wonder if this method would make sense in yarp::os::Connection

Reply:
I think that the connection is the "pipe" between two ports,  while the carrier specifies the type of data that passes through the pipe and the face is the network support on wich data relies on.
So the face dependes on carrier, and I think it is better mantain the face in the carrier.